packages/angular_devkit/build_webpack/src/webpack-dev-server/index2.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { BuilderContext, BuilderOutput, createBuilder } from '@angular-devkit/architect/src/index2';
+import { getSystemPath, json, normalize, resolve } from '@angular-devkit/core';
+import * as net from 'net';
+import { Observable, from, of } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import * as webpack from 'webpack';
+import * as WebpackDevServer from 'webpack-dev-server';
+import { ArchitectPlugin } from '../plugins/architect';
+import { WebpackFactory, WebpackLoggingCallback } from '../webpack/index2';
+import { Schema as WebpackDevServerBuilderSchema } from './schema';
+
+const webpackMerge = require('webpack-merge');
+
+
+export type DevServerBuildResult = BuilderOutput & {
+  port: number;
+  family: string;
+  address: string;
+};
+
+export function runWebpackDevServer(
+  config: webpack.Configuration,
+  context: BuilderContext,
+  options: {
+    devServerConfig?: WebpackDevServer.Configuration,
+    logging?: WebpackLoggingCallback,
+    webpackFactory?: WebpackFactory,
+  } = {},
+): Observable<BuilderOutput> {
+  const createWebpack = options.webpackFactory || (config => of(webpack(config)));
+  const log: WebpackLoggingCallback = options.logging
+    || ((stats, config) => context.logger.info(stats.toString(config.stats)));
+
+  config = webpackMerge(config, {
+    plugins: [
+      new ArchitectPlugin(context),
```

Comment:
Should this be added if progress reporting is not needed?

Reply:
Progress reporting should always be needed ;)

And yes, we might want to add more info to it later on.